### ISSUE TITLE
consolidate duplicate index.html code into json-utils.js ?

### ISSUE THREAD

**Issue by kfarr** [3DStreet/3dstreet#345]
lines 150-203 from 3dstreet repo are almost if not exactly identical to lines 183-235 in index.html from 3dstreet-editor repo

instead should all of these lines be in json-utils ?

```
 function inputStreetmix() {
      streetmixURL = prompt("Please enter a Streetmix URL", "https://streetmix.net/kfarr/3/example-street");
      setTimeout(function() { window.location.hash = streetmixURL; });
      streetContainerEl = document.getElementById('street-container');
      while (streetContainerEl.firstChild) {
        streetContainerEl.removeChild(streetContainerEl.lastChild);
      }
      streetContainerEl.innerHTML = '<a-entity street streetmix-loader="streetmixStreetURL: '+streetmixURL+'""></a-entity>';
    }

    // JSON loading starts here
    function getValidJSON(stringJSON) {
      // Preserve newlines, etc. - use valid JSON
      // Remove non-printable and other non-valid JSON characters
      return stringJSON.replace(/\'/g, "")
                       .replace(/\n/g, "")
                       .replace(/[\u0000-\u0019]+/g,"");
    }
    
    function createElementsFromJSON(streetJSON) {
      let streetObject = {};
      if (typeof streetJSON == 'string') {
        const validJSONString = getValidJSON(streetJSON);
        streetObject = JSON.parse(validJSONString);
      } else if (typeof streetJSON == 'object') {
        streetObject = streetJSON;
      }
      
      streetContainerEl = document.getElementById('street-container');
      while (streetContainerEl.firstChild) {
        streetContainerEl.removeChild(streetContainerEl.lastChild);
      }
      
      createEntities(streetObject.data, streetContainerEl);
      AFRAME.scenes[0].components['notify'].message('Scene loaded from JSON', 'success')
    }

    function inputJSON() {
      const stringJSON = prompt("Please paste 3DStreet JSON string");
      if (stringJSON) {
        createElementsFromJSON(stringJSON);
      }      
    }

    function fileJSON() {
      let reader=new FileReader();
      reader.onload=function(){
        createElementsFromJSON(reader.result);
      }
      reader.readAsText(this.files[0]);
    }

    document.getElementById('inputfile')
            .addEventListener('change', fileJSON);
```

**Comment by Algorush** [comment]
Yes, it is a good idea. Only functions with prompt (inputJSON, inputStreetmix) and function with FilerReader (fileJSON) are more about UI, not about utils? But maybe its ok to move them in one place

**Comment by kfarr** [comment]
closed by #346
